src/applications/lgy/coe/components/CoeDocumentUpload.jsx

```diff
@@ -10,7 +132,65 @@ export const CoeDocumentUpload = () => {
         COE application. Please send us all the documents listed so we can make
         a decision about your application.
       </p>
-      <FileInput buttonText="Upload this document" onChange={() => {}} />
+      {files.map((file, index) => (
+        <div
+          className="vads-u-background-color--gray-lightest vads-u-padding-y--1 vads-u-padding-x--2 vads-u-margin-y--1"
+          key={index}
+          id={index}
+        >
+          <p>
+            <strong>{file.name}</strong>
+          </p>
+          <button
+            onClick={() => onDeleteFile(index)}
+            className="usa-button-secondary vads-u-background-color--white vads-u-margin-top--0"
+          >
+            Delete file
+          </button>
+        </div>
+      ))}
+      <div
+        className={
+          documentType === 'Other'
+            ? 'vads-u-padding-left--1p5 vads-u-border-left--5px vads-u-border-color--primary-alt-light'
+            : null
+        }
+      >
+        <Select
+          required
+          name="document_type"
+          label="Select a document to upload"
+          options={DOCUMENT_TYPES}
+          onValueChange={onSelectChange}
+          value={{ dirty: false, value: documentType }}
+        />
+        {documentType === 'Other' && (
+          <TextInput
+            label="Document description"
+            name="document_description"
+            field={documentDescription}
+            required={documentType === 'Other'}
+            errorMessage={
+              validateIfDirty(documentDescription, isNotBlank)
+                ? null
+                : 'Please provide a description'
+            }
+            onValueChange={onTextInputValueChange}
+          />
+        )}
+      </div>
+      <FileInput
+        additionalClass={`${errorMsgClass} ${disabledOnEmptyDescClass}`}
```

Comment:
Currently in addition to providing error messaging on the text input above, we're using css to prevent pointer events on the file input if text input is blank. It isn't ideal, but currently there isn't a good way to force validation on the document description text input if a user leaves it blank and clicks on the file input to open the system dialog. This PR is already getting a little big, so I might explore other solutions in a future pull request, since there is still additional work to be done.